File: solutions/python/word-count/1/word_count.py
import re
from collections import Counter
def count_words(sentence):
    sentence = sentence.replace("_"," ")
    words = re.findall(r"\b(?:\w+(?:'\w+)?(?:_\w+)?|\w+)\b", sentence.lower())
    word_counts = Counter(words)
    return word_counts

File: solutions/javascript/word-count/1/word-count.js
//
//  Separators: : ! ? \t \n \s
//

export const countWords = (text) => {
  const countWords = {};
  const words = text.replace(/[&@$%^]/g, "").split(/[:!?\t\n\s,. ]+/g);
  words.forEach((w) => {
    const word = w.replace(/^'+|'+$/g, "").toLowerCase();
    if (!word) return;
    countWords[word] = (countWords[word] ?? 0) + 1;
  });
  return countWords;
};

console.log(countWords("Joe can't tell between 'large' and large."));
